contrib/download-frozen-image-v2.sh

```diff
@@ -205,62 +205,77 @@ get_target_arch() {
 		return 0
 	fi
 
-	if type go > /dev/null; then
-		go env GOARCH
-		return 0
-	fi
-
-	if type dpkg > /dev/null; then
-		debArch="$(dpkg --print-architecture)"
-		case "${debArch}" in
-			armel | armhf)
-				echo "arm"
+	if type uname > /dev/null; then
+		case "$(uname -m)" in
+			"x86_64")
+				echo amd64
 				return 0
 				;;
-			*64el)
-				echo "${debArch%el}le"
+			"i386")
+				echo 386
 				return 0
 				;;
-			*)
-				echo "${debArch}"
+			"aarch64")
+				echo arm64
 				return 0
 				;;
-		esac
-	fi
-
-	if type uname > /dev/null; then
-		uArch="$(uname -m)"
-		case "${uArch}" in
-			x86_64)
-				echo amd64
+			"arm64")
+				echo arm64
 				return 0
 				;;
-			arm | armv[0-9]*)
+			"armv*")
 				echo arm
+				;;
+			"mips")
+				echo mips
 				return 0
 				;;
-			aarch64)
-				echo arm64
+			"mipsle")
+				echo mipsle
 				return 0
 				;;
-			mips*)
-				echo >&2 "I see you are running on mips but I don't know how to determine endianness yet, so I cannot select a correct arch to fetch."
-				echo >&2 "Consider installing \"go\" on the system which I can use to determine the correct arch or specify it explicitly by setting TARGETARCH"
-				exit 1
+			"mips64")
+				echo mips64
+				return 0
 				;;
-			*)
-				echo "${uArch}"
+			"mips64le")
+				echo mips64le
+				return 0
```

Comment:
Feels like all of these _literal_ matches would already have been covered by the `*)`. I realise including them all makes it more explicit, but wondering if it's really needed (things would fail either way if it's a bogus architecture)

Reply:
I would be ok if this script was exclusively used in the Dockerfile. Maybe @cpuguy83 or @tianon could tell.